README.md

```diff
@@ -32,18 +36,30 @@ tinywav_close_write(&tw);
 ```C
 #include "tinywav.h"
 
-#define NUM_CHANNELS 1
```

Comment:
To promote good programing practice (IMO) I would write this as:
```
#include "tinywav.h"

#define NUM_CHANNELS 2
#define SAMPLE_RATE 48000
#define BLOCK_SIZE 480

TinyWav tw;
tinywav_open_read(&tw, 
	"path/to/input.wav",
	TW_SPLIT // the samples will be read into a split buffer
);

// samples are always provided in float32 format, 
// regardless of file sample format
float samples[NUM_CHANNELS * BLOCK_SIZE];
  
// Split buffer requires pointers to channel buffers
float* samplePtrs[[NUM_CHANNELS];
for (int j = 0; j < NUM_CHANNELS; ++j) {
  samplePtrs[j] = samples + j*BLOCK_SIZE;
}

for (int i = 0; i < 100; i++) {
  tinywav_read_f(&tw, samplePtrs, BLOCK_SIZE);
  
  ...
}

tinywav_close_read(&tw);
```

But maybe I just really like pointer arithmetic.

Reply:
Hmmm, isn't what you're proposing actually an example for `TW_INLINE` in terms of storage?

For a split buffer configuration, we have to assume the data for each channel is at a potentially completely different position in memory.

For example for a stereo signal, we could have:
```c
float samplesL[BLOCK_SIZE];
float samplesR[BLOCK_SIZE];
float* samplePtrs[] = { &samplesL[0], &samplesR[0] };
```

When I generalized this, I ended up with a 2D array.
But the fact that C stores 2D array contiguously in memory (same layout as a 1D array) is just a coincidence here...  I could have uses any multi-dimensional data structure instead, where this would not be the case.

Do you see what I mean?